src/Domain/Service/NoticeAssembler.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace App\Domain\Service;
+
+use App\DTO\Contribution;
+use App\Entity\Contributor;
+use App\Entity\DomainName;
+use App\Entity\MatchingContext;
+use App\Entity\Notice;
+use App\Helper\NoticeVisibility;
+use App\Repository\ContributorRepository;
+use App\Repository\DomainNameRepository;
+use InvalidArgumentException;
+
+class NoticeAssembler
+{
+    private $contributorRepository;
+    private $domainNameRepository;
+
+    public function __construct(ContributorRepository $contributorRepository, DomainNameRepository $domainNameRepository)
+    {
+        $this->contributorRepository = $contributorRepository;
+        $this->domainNameRepository = $domainNameRepository;
+    }
+
+    public function assemble(Contribution $contribution): Notice
+    {
+        return self::assembleNotice($contribution)
+            ->setContributor(self::assembleContributor($contribution))
+            ->addMatchingContext(self::assembleMatchingContext($contribution))
+        ;
+    }
+
+    public function assembleContributor(Contribution $contribution): Contributor
+    {
+        if ($contribution->isAQuestion()) {
+            if ($contribution->getToContributorId()) {
+                /** @var Contributor $questionedContributor */
+                $questionedContributor = $this->contributorRepository->find($contribution->getToContributorId());
+
+                if (!$questionedContributor) {
+                    throw new InvalidArgumentException("Questioned contributor doesn't exist.");
+                }
+
+                return $questionedContributor;
+            }
```

Comment:
I think `$contribution->isAQuestion()` but not `$contribution->getToContributorId()` we should either throw/reject or return a null contributor … 

But here we use the questionner as contributor (and may create it), which feels wrong to me for a question …

Reply:
I understand your concern, I'm trying to handle every cases here.

From the extension perspective, a *user* may ask a __question__ without asking it to any specific contributor so we shouldn't throw an error in such case. Well given we wan't to keep this feature @dis-moi/product 

I'll try to go down the path setting the `$contributor` to `null`. But as I'm typing this, I feel even more uncomfortable not having any kind validation... but let's say it's a temporary thing.

In fact it just highlight 2 facts:
- we badly need an authentication API.
- the "question" domain isn't clear enough

Reply:
I follow you … but a `null` `$contributor` does not feel too uncomfortable to me since it reflects what we really have : a future `Notice` without a `Contributor` yet 

But no blocker here !

Reply:
@lutangar @JalilArfaoui I say let's ship it and then add the authentication API later. Agreed, this is important to have, but no use solving problems we don't have yet. Let's get that auth API done quickly once this is done.

@lutangar are we okay to merge otherwise?

Reply:
@christpet 

Just discussed authentication with @Goutte and we are pretty confident it can be implemented swiftly.
Could you create a task for him going this way?

About this PR, I'll need to tackle the last few concerns first.